tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -87,398 +81,274 @@ var _ = Describe("odo devfile create command tests", func() {
-				helper.Cmd("odo", "create", "nodejs").ShouldPass()
-			})
+				// read the contents from the kubeconfig and replace the server entries
+				reg := regexp.MustCompile(`server: .*`)
+				kubeConfigContents, err := helper.ReadFile(path)
+				Expect(err).To(BeNil())
+				kubeConfigContents = reg.ReplaceAllString(kubeConfigContents, "server: https://not-reachable.com:443")
 
-			It("should successfully create the devfile component with --devfile points to the same devfile", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", "./devfile.yaml").ShouldPass()
-				fileIsEmpty, err := helper.FileIsEmpty("./devfile.yaml")
-				Expect(err).Should(BeNil())
-				Expect(fileIsEmpty).Should(BeFalse())
-			})
+				// write to a new file which will be used as the new kubeconfig
+				newKubeConfigPath = filepath.Join(commonVar.Context, "newKUBECONFIG")
+				newKubeConfig, err := os.Create(newKubeConfigPath)
+				Expect(err).To(BeNil())
+				defer newKubeConfig.Close()
 
-			It("should fail to create the devfile component with more than 1 arguments are passed in", func() {
-				helper.Cmd("odo", "create", "nodejs", "nodejs").ShouldFail()
+				_, err = newKubeConfig.WriteString(kubeConfigContents)
+				Expect(err).To(BeNil())
 			})
 
-			It("should fail to create the devfile component with --devfile points to different devfile", func() {
-				helper.Cmd("odo", "create", "nodejs", "--devfile", "/path/to/file").ShouldFail()
+			AfterEach(func() {
+				os.Remove(newKubeConfigPath)
 			})
 
-			It("should fail when we create the devfile component multiple times", func() {
-				helper.Cmd("odo", "create", "nodejs").ShouldPass()
-				output := helper.Cmd("odo", "create", "nodejs").ShouldFail().Err()
-				Expect(output).To(ContainSubstring("this directory already contains a component"))
+			It("should successfully create the devfile component and show json output", func() {
+				output := helper.Cmd("odo", "create", "nodejs", "--context", newContext, "-o", "json").WithEnv("KUBECONFIG="+newKubeConfigPath, "GLOBALODOCONFIG="+os.Getenv("GLOBALODOCONFIG")).ShouldPass().Out()
+				values := gjson.GetMany(output, "kind", "metadata.name", "status.state")
+				Expect(helper.GjsonMatcher(values, []string{"Component", "nodejs", "Unknown"})).To(Equal(true))
 			})
 		})
+	})
 
-		Context("Testing Create for OpenShift specific scenarios", func() {
-			JustBeforeEach(func() {
-				if os.Getenv("KUBERNETES") == "true" {
-					Skip("This is a OpenShift specific scenario, skipping")
-				}
-			})
-
-			It("should fail when we create the devfile", func() {
-				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), filepath.Join(commonVar.Context, devfile))
-				helper.Cmd("odo", "create", "nodejs").ShouldPass()
-				output := helper.Cmd("odo", "create", "nodejs").ShouldFail().Err()
-				Expect(output).To(ContainSubstring("this directory already contains a component"))
-			})
+	Context("odo create is executed with the --now flag", func() {
```

Comment:
You probably can merge all the Context in a single test (without having any BeforeEach).